### PR TITLE
[Workflow] Enforce `TODO_UPNEXT` comments to have an associated username

### PR DESCRIPTION
**Pull request by Olshansk**
## Summary

Improve process for `TODO_UPNEXT`

Goal is to block PRs on this: `TODO_UPNEXT: <do something>`
Allow PRs on this: `TODO_UPNEXT(@olshansk): <do something>`

Verification:

Screenshot 2024-10-28 at 5 18 20 PM

## Issue

NA

## Type of change

Select one or more from the following:

- [x] New feature, functionality or library
- [ ] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

### CODE REVIEW

**Comment by Olshansk** [comment]
@okdas Is there easy way to test this without merging it in?

File: .github/workflows/reviewdog.yml [hunk 1]
```diff
@@ -22,7 +22,25 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: pokt-network/action-fail-on-found@v1
+        with:
+          github_token: ${{ secrets.github_token }}
+          reporter: github-pr-review
+          level: error
+          fail_on_error: true
+          pattern: TODO_UPNEXT:
```
Comment (okdas) [inline .github/workflows/reviewdog.yml:43]:
```suggestion
          pattern: "TODO_UPNEXT:"
```
This line breaks the config: https://github.com/pokt-network/poktroll/actions/runs/11561203740/workflow

With this suggestion committed, I think this should work.

**Review by okdas — CHANGES_REQUESTED** [review]
> @okdas Is there easy way to test this without merging it in?

Yes - this PR can be used for testing. Just add `TODO_UPNEXT:` to any file.

**Review by bryanchriswhite — APPROVED** [review]
Very cool. :sunglasses: 

It would be nice if we could modify the comment contents to explain WHY it's complaining and what the author needs to change (i.e. add attribution). Perhaps it's out of scope but I'm wondering if that could be trivially possible via a different "reporter" or something?

(cc @okdas)

**Comment by Olshansk** [comment]
CCing @okdas and snoozing this email until next week.

**Pull request merged**